lambdas/functions/weka_api/api.go

```diff
@@ -0,0 +1,108 @@
+package weka_api
+
+import (
+	"context"
+	"fmt"
+	"github.com/rs/zerolog/log"
+	"github.com/weka/aws-tf/modules/deploy_weka/lambdas/common"
+	"os"
+	"strconv"
+
+	"github.com/weka/go-cloud-lib/logging"
+)
+
+func enrich(req *WekaApiRequest) error {
+	if req.Method == "" {
+		return fmt.Errorf("enrich> weka-api method is not set")
+	}
+	clusterName := os.Getenv("CLUSTER_NAME")
+	if clusterName == "" {
+		return fmt.Errorf("CLUSTER_NAME is not set")
+	}
+	ips, err := common.GetBackendsPrivateIps(clusterName, "backend")
+	if err != nil {
+		return fmt.Errorf("enrich > GetBackendsPrivateIps: %w", err)
+	}
+	log.Info().Msgf("enrich > Backend private IPs: %v", ips)
+	req.BackendPrivateIps = ips
+
+	req.managementLambdaName = os.Getenv("MANAGEMENT_LAMBDA")
+	if req.managementLambdaName == "" {
+		return fmt.Errorf("MANAGEMENT_LAMBDA is not set")
+	}
+
+	useSecretManagerEndpoint, err := strconv.ParseBool(os.Getenv("USE_SECRETMANAGER_ENDPOINT"))
+	if err != nil {
+		log.Warn().Msg("Failed to parse USE_SECRETMANAGER_ENDPOINT, assuming false")
+	}
+	if !useSecretManagerEndpoint {
```

Comment:
The ips fetch, shouldn't be called by the management.
When the secret endpoint isn't set, it means that the management lambda will no be able to fetch it, so the function which calls is should provide it to the the management. 
For the case where the secret is set, we already have the code inside the management function: https://github.com/weka/terraform-aws-weka/pull/308/files#diff-d3cb9141b1ffb3da0d19fb94e4d6d444a08154e6dda81d1af002d2d61cbf27adR435
So `fetch` should be called by status, or any other function that calls the management, before the actual call.
i.e. this can be replaced by generic `SetCreds`: https://github.com/weka/terraform-aws-weka/pull/308/files#diff-d3cb9141b1ffb3da0d19fb94e4d6d444a08154e6dda81d1af002d2d61cbf27adR340

Reply:
I'm not sure I understand you.
You mean that the management handler should not perform any logic to retrieve ips or set creds?

because the code you highlighted here happens on `WekaApiHandler` before the management function is invoked.

I can remove the code that existed for the `status` option on the management handler, so we only use the creds and ips already provided for any method.

Reply:
Management will fetch secrets only when the endpoint is set and this code is already exists.

Reply:
ecce9d95b6c48d699be8123f3f7a82e3d32423b3

deleted the fetch of the secrets endpoint on weka-api